src/chronify/sqlalchemy/functions.py

```diff
@@ -0,0 +1,33 @@
+import pandas as pd
+import polars as pl
+from numpy.dtypes import ObjectDType
+from pandas import DatetimeTZDtype
+from sqlalchemy import Connection, Selectable
+
+from chronify.models import TableSchema
+from chronify.time_configs import DatetimeRange
+
+
+def read_database(query: Selectable | str, conn: Connection, schema: TableSchema) -> pd.DataFrame:
+    """Read a database query into a Pandas DataFrame."""
+    df = pl.read_database(query, connection=conn).to_pandas()
+    config = schema.time_config
+    if config.needs_utc_conversion(conn.engine.name):
```

Comment:
Note SQLite special cases here and in the next function.

Reply:
Do we want to add pandas just to do the datetime conversion? Maybe we can just make simple function that uses bare python

Reply:
I think we agreed that pandas is part of the public API. It is what the user gets when they call `read_table`. Numpy array is not a good option because the table will have mixed types. Arrow table is an option, but that's not what users want. I'm skeptical that people want a duckdb relation. Thoughts?